filebeat/module/elasticsearch/deprecation/ingest/pipeline-json-8.yml

```diff
@@ -0,0 +1,15 @@
+description: Pipeline for parsing the Elasticsearch deprecation log file in JSON format.
+on_failure:
+- set:
+    field: error.message
+    value: '{{ _ingest.on_failure_message }}'
+processors:
+- json:
+    field: message
+    add_to_root: true
+- dot_expander:
+    field: '*'
+    override: true
+- set:
+    field: event.dataset
+    value: elasticsearch.deprecation
```

Comment:
Elasticsearch currently produces an inconsistent dataset for deprecation logs which has the product and type reversed (ie `deprecation.elasticsearch`). This is a temporary pipeline override that will be removed once the inconsistency is fixed on the producer side

Reply:
I think there was some intention around this https://github.com/elastic/elasticsearch/pull/68737 but lets not hold off this PR on this and follow up on it.